### PR TITLE
Update support for React Native

### PR DESCRIPTION
**Pull request by jpdriver**
Fix #1339

- [x] update `@types/react-native` dependency
- [x] add SafeAreaView to React Native interface
- [x] remove Navigator from React Native interface

### CODE REVIEW

**Comment by philpl** [comment]
@jpdriver It seems that our `@types/react-native` doesn't contain typings for this component yet. Can you upgrade the typings package please? 😁

> native/index.d.ts(51,55): error TS2694: Namespace '"/home/travis/build/styled-components/styled-components/node_modules/@types/react-native/index"' has no exported member 'SafeAreaView'.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ All notable changes to this project will be documented in this file. If a contri
 
 ## Unreleased
 
-- n/a
+- Add support for SafeAreaView when using styled-components in a React Native project
```
Comment (philpl) [inline CHANGELOG.md]: Can you add a small note linking to this PR please? :) `(see [#1339](https://github.com/styled-components/styled-components/pull/1339))`

**Comment by philpl** [comment]
cc @mxstbr We might want to bump this change into a minor release, since it removes `Navigator` which might break code that uses older versions of `react-native`

**Comment by mxstbr** [comment]
> since it removes Navigator which might break code that uses older versions of react-native

`styled-components@3.0.0`, that's a breaking change, let's ship that as a major version.

**Comment by philpl** [comment]
@mxstbr I'm a bit reluctant since previously we haven't used strict semver, and have instead used versions to convey major feature and maturity changes. So if we'd release this as v3 then we need to convey that that doesn't reflect feature/stability developments anymore.

Alternatively we could only alter `index.d.ts` to remove the outdated typings but leave the deprecated `Navigator` inside `native/index.js` for now

**Comment by brunolemos** [comment]
I arrived here looking for `SafeAreaView` support, maybe the PR should be splited to include this on v2 without the breaking change

**Comment by philpl** [comment]
@brunolemos definitely 👍 but I’m not convinced that there’s no alternative to removing Navigator altogether

**Comment by mxstbr** [comment]
We should just use SemVer.

**Comment by mxstbr** [comment]
> We should merge this, bump major, and put out a release note detailing when to update and what it entails. Agreed?

Let's ship this and #588 as v3.0.0? That seems like a logical grouping of changes, if you want to upgrade to newer ReactNative support you have to upgrade to `styled-components-native`? (I'll be taking a crack at #588 sometime soon hopefully)

**Comment by philpl** [comment]
@mxstbr that makes even more sense in my opinion. It'd particularly help with `no-parser` support in the future to have segregated packages. :+1:

@jpdriver sorry to put this on hold. I'll buy you a coffee for the trouble 😛

**Comment by morajabi** [comment]
> I'll buy you a coffee for the trouble 😛

You need to post a selfie with that coffee here. Also, you need to add "buying coffee for the awaited contributors" to the Core team members *responsibility* 😛

**Comment by jpdriver** [comment]
> I'll buy you a coffee for the trouble

he's a man of his word!

img_0371

**Comment by mxstbr** [comment]
Made my day! :laughing:

**Comment by morajabi** [comment]
OMG! Phil yo the man! I'm looking forward to seeing you in person 🤑🤑

@jpdriver You must buy me a coffee too, I tried so hard to convince him 😛 😛

**Comment by philpl** [comment]
@mxstbr DC&SA has a more fruity coffe though 😆 hate to break your heart though 💟

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Pull request merged**